src/utils/uint.ts

```diff
@@ -1,27 +1,53 @@
-import { InvalidTypeError } from '../errors/InvalidTypeError';
+import type {
+  Bytes32,
+  Bytes8,
+  BytesHex,
+  BytesHexNo0x,
+  Hex,
+  Uint,
+  Uint128,
+  Uint16,
+  Uint256,
+  Uint32,
+  Uint64,
+  Uint8,
+  UintBigInt,
+  UintNumber,
+} from '../types/primitives';
+import type {
+  RecordWithPropertyType,
+  RecordUintPropertyType,
+  RecordUint256PropertyType,
+} from './private';
+import { isRecordNonNullableProperty, typeofProperty } from './record';
 import { InvalidPropertyError } from '../errors/InvalidPropertyError';
-import { isNonNullableRecordProperty, typeofProperty } from './record';
-import type { Bytes32, Bytes8, BytesHexNo0x } from '../types/primitives';
-
-type UintNumber = number;
-type UintBigInt = bigint;
-type Uint = UintNumber | UintBigInt;
-type Uint8 = UintNumber | UintBigInt;
-type Uint32 = UintNumber | UintBigInt;
-type Uint64 = UintNumber | UintBigInt;
-type Uint256 = UintNumber | UintBigInt;
-
-export const MAX_UINT64 = BigInt('18446744073709551615'); // 2^64 - 1
+import { InvalidTypeError } from '../errors/InvalidTypeError';
+
+////////////////////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////////////////////
+
+// 2^8 - 1 = 255
+export const MAX_UINT8 = 0xff;
+
+// 2^16 - 1 = 65535
+export const MAX_UINT16 = 0xffff;
+
+// 2^32 - 1 = 4294967295
+export const MAX_UINT32 = 0xffffffff;
+
+// 2^64 - 1 = 18446744073709551615
+export const MAX_UINT64 = BigInt('0xffffffffffffffff');
```

Comment:
Use the `n` notation as in helper functions of this file?

```suggestion
export const MAX_UINT64 = 0xffffffffffffffffn;
```